renderdoc/driver/vulkan/vk_common.cpp

```diff
@@ -129,7 +129,8 @@ void GPUBuffer::Create(WrappedVulkan *driver, VkDevice dev, VkDeviceSize size, u
   device = dev;
   createFlags = flags;
 
-  align = (VkDeviceSize)driver->GetDeviceProps().limits.minUniformBufferOffsetAlignment;
+  align = std::max<VkDeviceSize>(driver->GetDeviceProps().limits.minUniformBufferOffsetAlignment,
+                                 driver->GetDeviceProps().limits.nonCoherentAtomSize);
```

Comment:
If I'm following this correctly, this should handle aligning the start of a map to `nonCoherentAtomSize`, but the size is still potentially mutable/non-aligned.

Would we need an additional align up in `GPUBuffer::Map` to ensure that `size` is a multiple of `align`? It should be fine just to change the size, since we already align the _next_ offset to a multiple of `align`.

Reply:
Apologies for the delay in responding. 

I agree that `size` also needs to be aligned. I am currently working on a patch to this PR that addresses this. 

Additionally, this first patch can increase the size of the allocation by effectively making each element occupy at least `nonCoherentAtomSize` bytes. This will also be addressed.

Reply:
I am abandoning this PR because there is a corrected version that will be included in a batch of changes to be submitted by @alexander-a-kharlamov.